src/main/java/emu/grasscutter/game/home/HomeModuleManager.java

```diff
@@ -143,15 +155,15 @@ public Either<List<GameItem>, Integer> claimAvatarRewards(int eventId) {
     }
 
     public Either<HomeAvatarSummonEvent, Integer> fireAvatarSummonEvent(
-            Player owner, int avatarId, int guid, int suiteId) {
+        Player owner, int avatarId, int guid, int suiteId) {
         var targetSuite =
-                ((HomeScene) owner.getScene())
-                        .getSceneItem().getBlockItems().values().stream()
-                                .map(HomeBlockItem::getSuiteList)
-                                .flatMap(Collection::stream)
-                                .filter(suite -> suite.getGuid() == guid)
-                                .findFirst()
-                                .orElse(null);
+            ((HomeScene) owner.getScene())
```

Comment:
i recommend using an `if (owner.getScene() instanceof HomeScene homeScene)` cast here; it ensures type safety as well as casting cleaner